typing/typecore.ml

```diff
@@ -2981,6 +3028,10 @@ let rec type_approx env sexp =
         raise(Error(sexp.pexp_loc, env, Expr_type_clash (trace, None, None)))
       end;
       ty2
+  | Pexp_apply
+      ({ pexp_desc = Pexp_extension({txt = "extension.unique"}, PStr []) },
+       [Nolabel, e]) ->
+    type_approx env e
```

Comment:
This is a pre-existing issue, but probably these cases should work for `ocaml.unique`, `unique`, `local` and `ocaml.local` too.